### ISSUE TITLE
PermanentRedirect?

### ISSUE THREAD

**Issue by tj** [Automattic/knox#178]
yo @domenic, updating from 0.5.x to 0.8.x and now im getting this for each request, I didn't notice anything that seemed related in the changelog, any idea?

```
<Error><Code>PermanentRedirect</Code><Message>The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.</Message><RequestId>86C86A0AFCB5795E</RequestId><Bucket>i.cloudup.com</Bucket><HostId>KowqHiRPuWlEG9o05aImJ3dMbzScbabmEW+ltRiaovKqrs6Hn0wvS15l07vdVZbw</HostId><Endpoint>i.cloudup.com.s3.amazonaws.com</Endpoint></Error>
```

**Comment by mihneasim** [comment]
I chose dev bucket to be Ireland, so it's closer to me. Thanks guys for clearing this redirection out. Still on in 0.9.2
I would have expected aws to write a Location header, but it doesn't. (i don't see it on request)

**Comment by PedroSena** [comment]
@AnthonyAkentiev When you say "The solution is to set 'region' option to 'eu-west-1'." Where should this be set ? I'm trying to do this on knox.createClient arguments but it does not work too

**Comment by AnthonyAkentiev** [comment]
@PedroSena, add region option like this:

     var client = knox.createClient({
          key: config.get('s3:key') 
          , secret: config.get('s3:secret')
          , bucket: config.get('s3:bucket')
          , region: config.get('s3:region')    // eu-west-1
     });

**Comment by PedroSena** [comment]
Thanks, I tried that but didn't work, still getting the redirect error, I'm migrating my bucket to a US Standard

**Comment by AnthonyAkentiev** [comment]
@PedroSena, please tell us if it will help you.

**Comment by PedroSena** [comment]
Yes, migrating the bucket to a US Standard one(was on Oregon before) worked, not the best path(had to migrate 30k files) but at least it is working now
